### PR TITLE
Switch to Webpack for UMD, many other structural changes.

### PR DESCRIPTION
**Pull request by Ambroos**
Major change in how Postscribe is built and structured. The resulting code hasn't really changed.

- Now using Webpack for packaging Postscribe as a UMD module. This means Postscribe is now usable with browser globals, CommonJS, AMD, …
- Replaced JSHint with ESLint. Does the same thing, but better.
- Cleaned up code according to stricter ESLint rules
- Bump version to Postscribe v2

**Structural changes**
- Moved `postscribe.js` to `/src/postscribe.js`
- Moved `htmlParser/htmlParser.js` to `/src/htmlParser.js`
- Moved helper functions to `/src/helpers.js`
- Moved `WriteStream` to `/src/WriteStream.js`
- Replaced the `set`-function with object-assign
- `htmlParser` is no longer exposed as a global (ever), but is available as `postscribe.htmlParser`.

**Changes in usage**
- htmlParser is no longer exported as a global, but it’s available under postscribe.htmlParser
- unpackaged files are no longer usable in the browser directly

The main reason for this is so Postscribe becomes easier to maintain and easier to use in modern Javascript projects. Webpack handles all module loading for you and makes Postscribe available as a UMD module. In addition to that, it means Postscribe could easily start using Babel in the future, split into separate NPM packages and make use of third-party scripts in an easier way. For a new client we make heavy use of Postscribe to support their ad platform, and we needed a version that's easier to integrate in our builds.

All tests are unchanged and still pass. For most users this should be a near-drop-in replacement. Because of the moved and restructured files all current pull requests will have to be updated before they can be merged again.

**For current users of Postscribe**
If you’re currently using Postscribe from the dist directory:
Nothing changes for you unless you directly call htmlParser. Replace all calls to `htmlParser` with calls to `postscribe.htmlParser` instead.

If you’re currently using Postscribe from the uncompiled source:
Don’t include `htmlParser.js` manually, use `postscribe(.min).js` from the dist directory. If you made direct calls to htmlParser, replace them with calls to `postscribe.htmlParser` instead.

*If anything has to be changed prior to merging, I'll gladly update this pull request.*
*This pull request contains a full dist build and version bump in package.json.*

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -83,10 +105,10 @@ module.exports = function(grunt) {
     });
   });
 
-  // Alias test
-  grunt.registerTask('test', ['generate_expected', 'qunit']);
-  grunt.registerTask('min', ['concat', 'uglify']);
+  grunt.registerTask('test', ['build', 'generate_expected', 'qunit']);
```
Comment (jnewman) [inline Gruntfile.js]: This was renamed to generateExpected, right?

**Comment by jnewman** [comment]
Thanks for this. I've not been available lately, but happy to dig into this on the weekend.

**Comment by Ambroos** [comment]
The issue with `generate_expected` was some confusion on my part about the Gruntfile, I just made it a bit more consistent. I added the `camelCase`-rule to the ESLint-config since the rest of the code adheres to it, so made that change to the Gruntfile too, but should've taken it further.

File: src/htmlParser.js [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+// Based on http://ejohn.org/blog/pure-javascript-html-parser/
+// Turns an HTML string into a token iterator.
+
+//TODO(#39)
+/*globals console:false*/
+var supports = (function() {
+  var target = {};
+
+  var html;
+  var work = this.document.createElement('div');
```
Comment (azaharakis) [inline src/htmlParser.js]: I've attempted to include this fork in my webpack project, and i'm running issues on this line. 
I'm assuming the intent here is for the window object to be passed in as the this context. however I don't see this done anywhere

Comment (Ambroos) [inline src/htmlParser.js]: I changed this to just use the global document. Can you check if it works?

Comment (azaharakis) [inline src/htmlParser.js]: This fixes my issue and loads postscribe correctly in my webpack package! Thanks! +1 for getting this merged and released asap!

**Comment by bra1n** [comment]
It would be great if this PR could be merged. What's holding me back from using Postscribe in my project so far is the lack of AMD support. Not everybody likes to have libraries the have to use the global namespace. ;)

**Comment by sethyates** [comment]
We believe this is fixed in 2.0.0.  Please retest against 2.0.0 release and open a new issue following Issue Reporting guidelines in 2.0.0 if you believe this is still an issue.

**Pull request closed without merging**